javascriptlib/src/mill/javascriptlib/CreateReactAppModule.scala

```diff
@@ -0,0 +1,171 @@
+package mill.javascriptlib
+import mill._
+import os.*
+import mill.define.Target
+
+// create-react-app: https://create-react-app.dev/docs/documentation-intro
+trait CreateReactAppModule extends TypeScriptModule {
+  override def npmDeps: T[Seq[String]] = Task {
+    Seq(
+      "react@18.3.1",
+      "react-dom@18.3.1",
+      "react-scripts@5.0.1",
+      "typescript@^4.9.5",
+      "web-vitals@2.1.4",
+      "@types/node@^16.18.119",
+      "@types/react@^18.3.12",
+      "@types/react-dom@^18.3.1"
+    )
+  }
+
+  override def npmDevDeps: T[Seq[String]] = Task {
+    Seq(
+      "@testing-library/jest-dom@^5.17.0",
+      "@testing-library/react@^13.4.0",
+      "@testing-library/user-event@^13.5.0",
+      "@types/jest@^27.5.2",
+      "serve@12.0.1"
+    )
+  }
+
+  override def sources: Target[PathRef] = Task.Source(millSourcePath)
+
+  def setup: Task[Unit] = Task.Anon {
+    // copy src files
+    os.copy(sources().path, Task.dest, mergeFolders = true)
+
+    def buildDependencies(dependencies: Seq[String]): ujson.Obj = {
+      val dependenciesObj = dependencies.map { dep =>
+        val lastAtIndex = dep.lastIndexOf('@')
+        if (lastAtIndex == -1)
+          dep -> ujson.Str("latest") // No '@' found, default version to "latest"; you probably should always provide the dependency version
+        else {
+          val name = dep.substring(0, lastAtIndex)
+          val version = dep.substring(lastAtIndex + 1)
+          name -> ujson.Str(version)
+        }
+      }
+      ujson.Obj.from(dependenciesObj)
+    }
+
+    val allPaths = Seq(
+      "*" -> Task.dest / "node_modules",
+      "app/*" -> "app",
+      "typescript" -> Task.dest / "node_modules" / "typescript"
+    )
+
+    // mk tsconfig.json
+    os.write(
+      Task.dest / "tsconfig.json",
+      ujson.Obj(
```

Comment:
These JSON blobs should probably be exposed as top level `Task[ujson.Value]` for users to configure directly